### PR TITLE
Cherry-pick GitHub Actions to 5.3

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
In anticipation of a new release, I think it's best we try the new release process on the build to make sure it works.

```release-note
NONE
```

### CODE REVIEW

**Comment by amyblais** [comment]
> cc @amyblais making sure this is okay with you. It shouldn't affect the final build of the app, just the release process.

Sounds good!

**Review by toninis — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        with:
+          node-version-file: "package.json"
+          cache: "npm"
+          cache-dependency-path: package-lock.json
+      - name: release/optimize
+        shell: powershell
+        run: ./scripts/Makefile.ps1 optimize
+      - name: release/install-deps
+        shell: powershell
+        run: ./scripts/Makefile.ps1 install-deps
```
[the remark this answers is not in the document]
Comment (toninis) [reply .github/workflows/release.yaml:88]: I mean it worked the other time without those but it failed in the other pipeline . So better add everything in place in order to be sure . 
We will have to revisit and make the pipelines more flexible and reusable

[the remark this answers is not in the document]
Comment (devinbinnie) [reply .github/workflows/release.yaml:88]: Ahh right, I'll make a PR against master to add this and then cherry-pick over to this branch

[the remark this answers is not in the document]
Comment (devinbinnie) [reply .github/workflows/release.yaml:88]: https://github.com/mattermost/desktop/pull/2601

**Review by toninis — APPROVED** [review]
🚀

**Comment by toninis** [comment]
We will closely monitor during the release . @devinbinnie ping me in order to check together the progress when you trigger it

**Pull request merged**